compiler/main/config.cpp

```diff
@@ -68,40 +85,53 @@ void checkConfigs() {
 // the internal scope-less BlockStmt.
 
 void parseCmdLineConfig(const char* name, const char* value) {
-  // Generate a C-string for a nominal Chapel assignment statement
-  const char* stmtText = (value[0] != '\0') ? astr("dummyConfig=", value, ";") : astr("dummyConfig=true;");
-  const char* parseFn  = astr("Command-line arg (", name, ")");
-  const char* parseMsg = astr("parsing '", value, "'");
 
-  // Invoke the parser to generate AST
-  BlockStmt*  stmt     = parseString(stmtText, parseFn, parseMsg);
+  // unfortunately this is parsed in the order from the command line, so for
+  // it to work, --dyno must come before -sConfigVar=Val or it will not try
+  // to use the dyno parser for command line input
```

Comment:
Seems like something we could handle by adjusting `process_args` and related code but this is probably OK for now.

Reply:
eventually (in the not-too-distant future) this conditional will go away and it won't be a problem